datafusion/physical-expr/src/intervals/interval_aritmetic.rs

```diff
@@ -508,6 +580,83 @@ fn cast_scalar_value(
     ScalarValue::try_from_array(&cast_array, 0)
 }
 
+/// This function calculates the final cardinality result by inspecting the endpoints of the interval.
+fn calculate_cardinality_based_on_bounds(
+    lower_open: bool,
+    upper_open: bool,
+    diff: u64,
+) -> u64 {
+    match (lower_open, upper_open) {
+        (false, false) => diff + 1,
+        (true, true) => diff - 1,
+        _ => diff,
+    }
+}
+
+trait OneTrait: Sized + std::ops::Add + std::ops::Sub {
```

Comment:
Perhaps you could use ScalarValue::new_one here: https://docs.rs/datafusion/latest/datafusion/common/enum.ScalarValue.html#method.new_one